### PR TITLE
Connection: Load connection JS bundle via connection PHP package

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Extracted out of #38330 

### Problem

For publicize (Social), the JS code that gets bundled in Jetpack takes precedence over Social even if Social is the latest version. This is not the case with PHP packages, for which we always load the latest version regardless.

This mismatch of PHP and JS code often results in bugs.

Also, bundled JS code makes it difficult to fix a bug in Jetpack given the monthly release cycle. Thus, we decided to bundle the publicize JS code into publicize package to ensure that the latest version of the  publicize JS code is loaded both in Jetpack and Social.

Now, to be able to externalize publicize, we also need to externalize connection js code to avoid it getting bundled into publicize code and fix some API call errors due to duplicate instances of connection bundle both in publicize bundle as well as in the plugins (Jetpack and Social)

#### Benefits
- JS Code for connection will always be in sync with its PHP package, i.e. the latest version gets loaded.
- Plugins will have smaller js bundles

## Proposed changes:

* Bundle connection JS code inside connection PHP package
* Let connection PHP package handle the registration or script for the connection bundle
* Externalize that connection bundle for all the plugins/packages
* The connection bundle will now be available via `window.JetpackConnection` just like `@wordpress/*` packages are available via `window.wp.*`

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
pdrWKz-1Mq-p2

## Testing instructions:

* Run `jetpack build packages/assets packages/connection plugins/jetpack`
* Just do a smoke test of the pages that rely on Jetpack connection

### CODE REVIEW

**Comment by manzoorwanijk** [comment]
I will fix up versions once this gets an approval to avoid noise in the PR.

**Review by coder-karen — DISMISSED** [review]
👋 I gave this a spin, testing different scenarios related to connection. One thing I noticed was that both on disconnecting and connecting Jetpack, with this PR applied the related modals seemed to be missing styling. So it looks like the way CSS styles are applied may be affected in some way via this PR. Examples here:
Connection:
Screenshot 2024-08-19 at 10 53 01
Disconnection (bottom part of modal):
Screenshot 2024-08-19 at 10 56 46

**Comment by manzoorwanijk** [comment]
Thank you for giving this a test...

> 👋 I gave this a spin, testing different scenarios related to connection. One thing I noticed was that both on disconnecting and connecting Jetpack, with this PR applied the related modals seemed to be missing styling. So it looks like the way CSS styles are applied may be affected in some way via this PR.

Did you run `jetpack build packages/connection plugins/jetpack` before loading that page?

**Comment by manzoorwanijk** [comment]
> 👋 I gave this a spin, testing different scenarios related to connection. One thing I noticed was that both on disconnecting and connecting Jetpack, with this PR applied the related modals seemed to be missing styling. So it looks like the way CSS styles are applied may be affected in some way via this PR.

I have fixed the issue for connection screen. I assumed that all our scripts are registered via `Automattic\Jetpack\Assets::register_script` and enqueued via `Automattic\Jetpack\Assets::enqueue_script` which automatically takes care of loading the styles, but I noticed that it's not the case with admin page styles and thus had to explicitly add connection CSS as a dependency.

File: projects/packages/connection/webpack.config.js [hunk 1]
```diff
@@ -16,8 +16,57 @@ for ( const file of glob.sync( './src/sso/*.css' ) ) {
...
+				extraLoaders: [ 'sass-loader' ],
+			} ),
+			// Handle images.
+			jetpackWebpackConfig.FileRule(),
+		],
+	},
+	externals: {
+		...jetpackWebpackConfig.externals,
+		jetpackConfig: JSON.stringify( {
+			consumer_slug: 'identity_crisis',
```
Comment (anomiex) [inline projects/packages/connection/webpack.config.js]: We should probably update this to something like
```suggestion
			consumer_slug: 'connection_package',
```
Or move it out of the shared config and into the separate configs below.

**Comment by anomiex** [comment]
@coder-karen If the issues you found are fixed, I think this is good to go. Do you have time to check it again? Otherwise I'll try to test it.

**Review by anomiex — APPROVED** [review]

**Pull request merged**